src/aria/templates/ClassGenerator.js

```diff
@@ -113,16 +113,19 @@ module.exports = Aria.classDefinition({
          * parameter given to the callback function is: { classDef: {String} if null, errors occured during parsing or
          * class generation; otherwise contains the generated class }
          * @param {String} template the template
-         * @param {Boolean} allDeps If true, all dependencies should be included in the generated class. Otherwise only
-         * put classes which are not currently loaded in the $dependencies section.
+         * @param {aria.templates.CfgBeans:ClassGeneratorCfg} options Options for the class generation.
          * @param {aria.core.CfgBeans:Callback} callback the callback description
-         * @param {Object} context - passes template context information to improve debugging
-         * @param {Boolean} debug debug mode
          */
-        parseTemplate : function (template, allDeps, callback, context, debug, skipLogError) {
+        parseTemplate : function (template, optionsOrAllDeps, callback, context, debug, skipLogError) {
+            var options = (typeof optionsOrAllDeps == "object") ? optionsOrAllDeps : {
+                allDependencies: optionsOrAllDeps,
+                errorContext: context,
+                debug: debug,
+                skipLogError: skipLogError
+            };
             var tree, errors;
             try {
-                tree = this._parser.parseTemplate(template, context, this.STATEMENTS, skipLogError);
+                tree = this._parser.parseTemplate(template, options.errorContext, this.STATEMENTS, options.skipLogError);
```

Comment:
Since you use the `options` here you should maybe do the same thing as in `__buildClass`: validating the object. I understand that then there would be more work to avoid duplicate validation since this `__buildClass` method is called afterwards.